### ISSUE TITLE
tfjs-backend-webgpu multiple issues

### ISSUE THREAD

**Issue by vladmandic** [tensorflow/tfjs#5494]
i've tried using `tfjs-backend-webgpu` with 10 different models and got 4 different errors  
success rate so far is zero as each model resulted in at least one error

environment: chrome/94 canary, tfjs 3.8.0, tfjs-backend-webgpu 0.0.1-alpha.7

## this error is most common

```log
Tint Reflection failure:
Workgroup X dimension exceeds maximum allowed:512 > 256
    at ReflectShaderUsingTint (../../third_party/dawn/src/dawn_native/ShaderModule.cpp:637)
    at InitializeBase (../../third_party/dawn/src/dawn_native/ShaderModule.cpp:1238)
    at Create (../../third_party/dawn/src/dawn_native/d3d12/ShaderModuleD3D12.cpp:170)
    at GetOrCreateShaderModule (../../third_party/dawn/src/dawn_native/Device.cpp:703)
```

note that depending on model, numbers in the error message may be different
e.g. `1024 > 256` or `

## followed by this error

```log
Tint Reflection failure:
Workgroup shared storage size for main exceeds the maximum allowed: 32768 > 16352
    at ReflectShaderUsingTint (..<URL>)
    at InitializeBase (..<URL>)
    at Create (..<URL>)
    at GetOrCreateShaderModule (..<URL>)
```

same as with first error, numbers in the error message may be different depending on the model  
e.g. `16640 > 16352`

## this happens only on one model, `mb3-centernet`

```log
Parse failed
glslang.onefile.js:46 ERROR: 0:129: 'return' : type does not match, or is not convertible to, the function's return type 
glslang.onefile.js:9 Uncaught (in promise) Error: GLSL compilation failed
    at Object.d.compileGLSLZeroCopy (glslang.onefile.js:9)
    at compileProgram (webgpu_program.ts:75)
    at backend_webgpu.ts:723
    at WebGPUBackend.getAndSavePipeline (backend_webgpu.ts:444)
    at WebGPUBackend.runWebGPUProgram (backend_webgpu.ts:722)
```

## and this is a common error in preprocessing

(ok, this one is a missing feature, not a bug strictly speaking)

```log
engine.ts:548 Uncaught (in promise) Error: Kernel 'RotateWithOffset' not registered for backend 'webgpu'
  at Engine.runKernel (engine.ts:548)
  at rotateWithOffset_ (rotate_with_offset.ts:58)
  at Object.rotateWithOffset__op [as rotateWithOffset] (operation.ts:51)
```

cc @qjia7

**Comment by vladmandic** [comment]
closing as resolved when building `tfjs-backend-webgpu` from main branch,  
seems that latest released version `tfjs-backend-webgpu 0.0.1-alpha.7` is just too old

**Comment by qjia7** [comment]
@vladmandic Thanks for trying the webgpu backend. Sorry that the publish release one `0.0.1-alpha.7` hasn't been updated for a while. Recently, WebGPU changed a lot to reach the Chrome's origin trials (OT). So we did many changes to satisfy  them. So for now, we suggest build` tfjs-backend-webgpu` from main branch like you did. Currently, we still need 1~2 weeks to totally switch to use WGSL instead GLSL before releasing a new npm package.
And it's really appreciated that you reported the issues in webgpu and provided your feedbacks. We will follow up them and make sure the issues can be resolved as soon as possible. Please let us know if you meet any issues. Thanks.
